components/studentPortal/layout/useLastSeenModal.tsx

```diff
@@ -17,6 +22,7 @@ export const useLastSeenModal = (
   const [updateLastSeenModal] = useMutation(UPSERT_LAST_SEEN_MODAL);
   const [showOnboardingModal, setShowOnboardingModal] = useState(false);
   const [showExitModal, setShowExitModal] = useState(false);
+  const { email } = useSelector(profileSelector);
```

Comment:
this could be 
`const { user } = useAuth()`
`const email = user.email`

and you could get rid of all your redux stuff. There's no need for your redux changes, since we already have the users email inside the useAuth hook

Reply:
great idea! so much cleaner.